# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студенты гр. 2381 Соколов Сергей и Тищенко Артем. Вариант №22

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

**Интервальной** называется статистическая оценка, которая задается двумя числами $-$ концами интервала (в котором может находиться значение параметра $\Theta$).

**Точностью $\Theta^*$** статистической оценки параметра $\Theta$ называется значение $\delta$, удовлетворяющее неравенству:

$$\delta > \left|\Theta - \Theta^*\right|$$

Чем меньше $\delta$, тем точнее оценка $\Theta^*$

**Надежностью (доверительной вероятностью)** оценки называется вероятность $\gamma$, с которой выполняется неравенство:

$$\gamma = P\left\{\delta > \left|\Theta - \Theta^*\right|  \right\}$$

Доверительный интервал $-$ интервал $(\Theta^* - \delta; \Theta^* + δ)$, является интервальной оценкой параметра $\Theta$

**Статистической гипотезой** называют гипотезу о виде неизвестного распределения исследуемой случайной величины или о значении параметров известных распределений.

Выдвинутую гипотезу называют **нулевой или основной гипотезой и обозначают** $H_0$

**Конкурирующей или альтернативной** называют гипотезу $H_1$, которая противоречит основной гипотезе.

Различают **простые гипотезы** (содержат только одно предположение) и **сложные гипотезы** (содержат более одного предположения).

Выдвинутые статистические гипотезы могут быть верны или не верны. Поэтому возникает необходимость их проверки. При проверке могут иметь место ошибки первого и второго рода.

**Ошибка первого рода** состоит в том, что нулевая гипотеза отвергается, хотя она верна.
Вероятность совершения ошибки первого рода называют уровнем значимости и обозначают через $\alpha$

**Ошибка второго рода** состоит в том, что нулевая гипотеза принимается, хотя она неверна. Вероятность совершения ошибки второго рода обозначают через $\beta$

Величину $1 - \beta$ $-$ вероятность  того,что гипотеза отвергается, когда она неверна, называют мощностью критерия

Для проверки статистических гипотез предлагаются различные критерии, которые представляют собой значения случайных величин, имеющих те или иные известные законы распределения.
При проверке статистических гипотез на основе выборочных данных вычисляются значения этих случайных величин (**наблюдаемые значения**), которые сопоставляются с некоторыми, так называемыми, **критическими** (зависящими от заданного уровня значимости $\alpha$) **значениями** этих случайных величин. На основании этого сопоставления принимается решение о принятии или не принятии нулевой гипотезы.

**Областью принятия нулевой гипотезы** называют совокупность значений критерия, при которых нулевую гипотезу принимают.

**Критической областью** называют совокупность значений критерия, при которых нулевую гипотезу отвергают.

Область принятия гипотезы и критическая область разделены точками, которые называются **критическими значениями критерия** $K_{crit}$

## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2$. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы

Подключение библиотек, загрузка интервального ряда, полученного в ПР1, присвоение переменным значений:

$N $ $-$ размер выборки

$s $ $-$ исправленное СКО, получено в ПР2

$xv $ $-$ выборочное среднее, получено в ПР2

In [1]:
import numpy as np
import pandas as pd

N = 112
s = 54.92160019284322
xv = 457.0122767857141
df = pd.read_csv('is.csv', index_col=0)
df

,[xi; xi+1),mid_xi,mi,rel_mi,s_mi,s_rel_mi
0,[320.0; 353.875),336.9375,3,0.026786,3,0.026786
1,[353.875; 387.75),370.8125,8,0.071429,11,0.098214
2,[387.75; 421.625),404.6875,18,0.160714,29,0.258929
3,[421.625; 455.5),438.5625,28,0.250000,57,0.508929
4,[455.5; 489.375),472.4375,24,0.214286,81,0.723214
5,[489.375; 523.25),506.3125,18,0.160714,99,0.883929
6,[523.25; 557.125),540.1875,8,0.071429,107,0.955357
7,[557.125; 591.0],574.0625,5,0.044643,112,1.000000


#### Задание 1
Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности $γ \in \{0.95, 0.99\}$. Сделать выводы.

In [2]:
gamma = [0.95, 0.99]
# t полученно из Приложения 6 для заданных доверительных точностей и размера выборки 100
t = [1.984, 2.627]
left = []
right = []
delta = []
for i in range(len(gamma)):
  left.append(xv - t[i] * s / np.sqrt(N))
  right.append(xv + t[i] * s / np.sqrt(N))
  delta.append((right[i] - left[i]) / 2)
  print(f"Доверительный интервал с доверительной точностью {gamma[i]}: ({left[i]}; {right[i]}) \nТочность:{delta[i]}")

Доверительный интервал с доверительной точностью 0.95: (446.7161036035532; 467.308449967875) 
Точность:10.296173182160885
Доверительный интервал с доверительной точностью 0.99: (443.3791886055041; 470.6453649659241) 
Точность:13.633088180209995


При увеличении доверительной точности доверительный интервал и точность математического ожидания увеличились, следовательно точность оценки уменьшилась

#### Задание 2
Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение $q$ при заданных
$\gamma$ и $n$. Построить доверительные интервалы, сделать выводы.

In [3]:
# q полученно из Приложения 7 для заданных доверительных точностей и размера выборки 100
q = [0.143, 0.198]
sigma_left = []
sigma_right = []
for i in range(len(gamma)):
  sigma_left.append(s * (1 - q[i]))
  sigma_right.append(s * (1 + q[i]))
  print(f"Доверительный интервал СКО с точностью {gamma[i]}: ({sigma_left[i]}; {sigma_right[i]})")

Доверительный интервал СКО с точностью 0.95: (47.06781136526664; 62.7753890204198)
Доверительный интервал СКО с точностью 0.99: (44.047123354660265; 65.79607703102617)


При увеличении доверительной точности Доверительный интервал СКО увеличился, следовательно точность оценки СКО уменьшилась

#### Задание 3
Проверить гипотезу о нормальности заданного распределения с помощью критерия
$χ^2$ (Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить табл. 1.

##### Расчет теоретических частот

In [4]:
import scipy.integrate as integrate

df1 = pd.DataFrame(columns=['x_i', 'x_i+1', 'n_i', 'z_i', 'z_i+1', 'Phi(z_i)', 'Phi(z_i+1)', 'p_i', "n_i'"])
df1['n_i'] = df['mi']
for i in range(len(df['[xi; xi+1)'])):
  tmp = df.loc[i, '[xi; xi+1)'][1:-1].split(';')

  df1.loc[i, 'x_i'] = np.float64(tmp[0])
  df1.loc[i, 'x_i+1'] = np.float64(tmp[1])

  df1.loc[i, 'z_i'] = (df1.loc[i, 'x_i'] - xv) / s if i != 0 else -np.inf
  df1.loc[i, 'z_i+1'] = (df1.loc[i, 'x_i+1'] - xv) / s if i != len(df['[xi; xi+1)']) - 1 else np.inf

  df1.loc[i, 'Phi(z_i)'] = integrate.quad(lambda x: 1 / np.sqrt(2 * np.pi) * np.exp(-(x ** 2) / 2), 0, df1.loc[i, 'z_i'], epsabs=1.49e-16, epsrel=1.49e-14)[0]
  df1.loc[i, 'Phi(z_i+1)'] = integrate.quad(lambda x: 1 / np.sqrt(2 * np.pi) * np.exp(-(x ** 2) / 2), 0, df1.loc[i, 'z_i+1'], epsabs=1.49e-16, epsrel=1.49e-14)[0]

  df1.loc[i, 'p_i'] = df1.loc[i, 'Phi(z_i+1)'] - df1.loc[i, 'Phi(z_i)']
  df1.loc[i, "n_i'"] = N * df1.loc[i, 'p_i']
df1

,x_i,x_i+1,n_i,z_i,z_i+1,Phi(z_i),Phi(z_i+1),p_i,n_i'
0,320.0,353.875,3,-inf,-1.8779,-0.5,-0.469803,0.030197,3.382111
1,353.875,387.75,8,-1.8779,-1.261112,-0.469803,-0.396366,0.073437,8.22493
2,387.75,421.625,18,-1.261112,-0.644323,-0.396366,-0.240317,0.156049,17.477438
3,421.625,455.5,28,-0.644323,-0.027535,-0.240317,-0.010984,0.229334,25.685362
4,455.5,489.375,24,-0.027535,0.589253,-0.010984,0.222154,0.233138,26.111435
5,489.375,523.25,18,0.589253,1.206041,0.222154,0.386099,0.163945,18.361838
6,523.25,557.125,8,1.206041,1.82283,0.386099,0.465835,0.079736,8.930451
7,557.125,591.0,5,1.82283,inf,0.465835,0.5,0.034165,3.826435


###### Заполнение расчетной таблицы

In [5]:
df2 = pd.DataFrame(columns = ['[xi; xi+1)', 'n_i','p_i', 'n_i\'', '(n_i - n_i\')^2', '(n_i - n_i\')^2 / n_i\'',  'n_i^2', 'n_i^2 / n_i\''])
df2['[xi; xi+1)'] = df['[xi; xi+1)']
df2['n_i'] = df['mi']
df2['p_i'] = df1['p_i']
df2['n_i\''] = df1['n_i\'']
df2['(n_i - n_i\')^2'] = (df2['n_i'] - df2['n_i\'']) ** 2
df2["(n_i - n_i')^2 / n_i'"] = df2['(n_i - n_i\')^2'] / df2['n_i\'']
df2['n_i^2'] = df2['n_i'] ** 2
df2['n_i^2 / n_i\''] = df2['n_i^2'] / df2['n_i\'']

df2.loc['sum'] = df2.sum()
df2.loc['sum', '[xi; xi+1)'] = np.nan
df2.loc['sum', '(n_i - n_i\')^2'] = np.nan
df2.loc['sum', 'n_i^2'] = np.nan
df2

,[xi; xi+1),n_i,p_i,n_i',(n_i - n_i')^2,(n_i - n_i')^2 / n_i',n_i^2,n_i^2 / n_i'
0,[320.0; 353.875),3,0.030197,3.382111,0.146009,0.043171,9.0,2.66106
1,[353.875; 387.75),8,0.073437,8.22493,0.050593,0.006151,64.0,7.781222
2,[387.75; 421.625),18,0.156049,17.477438,0.273071,0.015624,324.0,18.538186
3,[421.625; 455.5),28,0.229334,25.685362,5.357548,0.208584,784.0,30.523222
4,[455.5; 489.375),24,0.233138,26.111435,4.458158,0.170736,576.0,22.059301
5,[489.375; 523.25),18,0.163945,18.361838,0.130927,0.00713,324.0,17.645292
6,[523.25; 557.125),8,0.079736,8.930451,0.865739,0.096942,64.0,7.166491
7,[557.125; 591.0],5,0.034165,3.826435,1.377256,0.359932,25.0,6.533497
sum,NaN,112,1.0,112.0,NaN,0.90827,NaN,112.90827


Получим таблицу:

|     | $[x_i$$;$ $x_{i+1})$        |   $n_i$ |       $p_i$ |      $n_i'$ | $(n_i - n_i')^2$      |   $\frac{(n_i - n_i')^2}{n_i'}$ | $n_i^2$   |   $\frac{n_i^2}{n_i'}$ |
|:----|:------------------|------:|----------:|----------:|:--------------------|------------------------:|:--------|---------------:|
| 1   | [320.0; 353.875)  |     3 | 0.0301974 |   3.38211 | 0.14600871729997675 |              0.0431709  | 9       |        2.66106 |
| 2   | [353.875; 387.75) |     8 | 0.0734369 |   8.22493 | 0.05059328587092079 |              0.00615121 | 64      |        7.78122 |
| 3   | [387.75; 421.625) |    18 | 0.156049  |  17.4774  | 0.2730707581705649  |              0.0156242  | 324     |       18.5382  |
| 4   | [421.625; 455.5)  |    28 | 0.229334  |  25.6854  | 5.357548358289772   |              0.208584   | 784     |       30.5232  |
| 5   | [455.5; 489.375)  |    24 | 0.233138  |  26.1114  | 4.458158380259412   |              0.170736   | 576     |       22.0593  |
| 6   | [489.375; 523.25) |    18 | 0.163945  |  18.3618  | 0.1309269809799354  |              0.00713039 | 324     |       17.6453  |
| 7   | [523.25; 557.125) |     8 | 0.0797362 |   8.93045 | 0.8657391729713085  |              0.0969424  | 64      |        7.16649 |
| 8   | [557.125; 591.0]  |     5 | 0.0341646 |   3.82643 | 1.3772556359242476  |              0.359932   | 25      |        6.5335  |
| $\sum$ | $-$                 |   112 | 1         | 112       | $-$                   |              0.90827    | $-$       |      112.90827   |



#### Задание 4
Доказать, что
$$\chi^2_{obs} = \overset{k}{\underset{i=1}{\sum}}\frac{n^2_i}{n'_i} - n$$

Проконтролировать корректность вычисления $\chi^2_{obs}$

$$n = \overset{k}{\underset{i=1}{\sum}}n_i$$

##### Доказательство:

$$\chi^2_{obs} = \overset{k}{\underset{i=1}{\sum}}\frac{(n_i - n_i')^2}{n'_i}$$

Рассмотрим разницу:

$$\overset{k}{\underset{i=1}{\sum}}\frac{(n_i - n_i')^2}{n'_i} - \left(\overset{k}{\underset{i=1}{\sum}}\frac{n^2_i}{n'_i} - n \right)= \\ = \overset{k}{\underset{i=1}{\sum}}\left(\frac{n^2_i}{n'_i} - 2n_i + n_i' - \frac{n^2_i}{n'_i}\right) + n = \\ = \overset{k}{\underset{i=1}{\sum}}n_i' - 2n  + n =  \overset{k}{\underset{i=1}{\sum}}n_i' - n$$

Рассмотрим $\overset{k}{\underset{i=1}{\sum}}n_i'$:

$$\overset{k}{\underset{i=1}{\sum}}n_i' = \overset{k}{\underset{i=1}{\sum}}np_i= n\overset{k}{\underset{i=1}{\sum}} \left(\Phi(z_{i+1}) - \Phi(z_i)\right) = \\ = n(\Phi(z_{k}) - \Phi(z_{k-1}) + \Phi(z_{k-1}) - \Phi(z_{k-2}) + \dots - \Phi(z_{1}))= \\ =n(\Phi(z_{k}) - Φ(z_{1}))= n(\Phi(∞) - Φ(-∞)) =\\ = n (0.5 - (-0.5)) = n \cdot 1 = n$$

Следовательно:

$$\overset{k}{\underset{i=1}{\sum}}n_i' - n = n -n = 0 ⇒\\ \overset{k}{\underset{i=1}{\sum}}\frac{(n_i - n_i')^2}{n'_i} - \left(\overset{k}{\underset{i=1}{\sum}}\frac{n^2_i}{n'_i} - n \right)= 0 ⇒ \\ ⇒ \overset{k}{\underset{i=1}{\sum}}\frac{(n_i - n_i')^2}{n'_i} = \overset{k}{\underset{i=1}{\sum}}\frac{n^2_i}{n'_i} - n ⇒ \\ ⇒ \chi^2_{obs}= \overset{k}{\underset{i=1}{\sum}}\frac{n^2_i}{n'_i} - n$$

Что и требовалось доказать

In [6]:
print(df2.loc['sum', "(n_i - n_i')^2 / n_i'"] - (df2.loc['sum', 'n_i^2 / n_i\'']) + N)

0.0


Разность значений критерия Пирсона, полученная двумя разными способами равна $0$. Следовательно значения $\chi^2_{obs}$ посчитанно корректно

#### Задание 5
По заданному уровню значимости $α = 0.05$ и числу степеней свободы $df$ найти критическую точку $\chi^2_{crit}$ и сравнить с наблюдаемым значением. Сделать выводы.

In [7]:
print(f'Число степеней свободы: {len(df['[xi; xi+1)']) - 3}')

Число степеней свободы: 5


Критическая точка $\chi^2_{crit} = 11.1 > \chi^2_{obs} = 0.90827$, следовательно гипотиза о нормальном распределении случайной велечины принимается.

## Выводы
Были вычислены интервальные статистические оценки математического ожидания и среднеквадратичного отклонения. Была проверена гипотеза о нормальном распределении выборки с помощью критерия Пирсона $\chi^2$, после проверки гипотеза была принята.